interface/login/login.php

```diff
@@ -40,6 +41,59 @@
 $twig = new TwigContainer(null, $GLOBALS["kernel"]);
 $t = $twig->getTwig();
 
+/**
+ * @var Finder
+ */
+$primaryFinder = new Finder();
+$basePrimaryLoginLogoDir = $GLOBALS['OE_SITE_DIR'] . "/images/logos/core/login/primary";
+$primaryFinder->files()->in($basePrimaryLoginLogoDir)->name("login-logo.*");
+
+if ($primaryFinder->hasResults()) {
+    foreach ($primaryFinder as $f) {
+        $primaryLoginLogoPath = $GLOBALS['OE_SITE_WEBROOT'] . "/images/logos/core/login/primary/" . $f->getFileName();
+    }
+} else {
+    $primaryLoginLogoPath = $GLOBALS['images_static_relative'] . "/login-logo.png";
+}
+
+$secondaryFinder = new Finder();
+$baseSecondaryLoginLogoDir = $GLOBALS['OE_SITE_DIR'] . "/images/logos/core/login/secondary";
+$secondaryFinder->files()->in($baseSecondaryLoginLogoDir)->name("login-logo.*");
+
+if ($secondaryFinder->hasResults()) {
+    foreach ($secondaryFinder as $f) {
+        $secondaryLoginLogoPath = $GLOBALS['OE_SITE_WEBROOT'] . "/images/logos/core/login/secondary/" . $f->getFileName();
+    }
+} else {
+    $secondaryLoginLogoPath = $GLOBALS['images_static_relative'] . "/login-logo.png";
+}
+
+$smallLogoFinder = new Finder();
+$baseSmallLogoOneDir = $GLOBALS['OE_SITE_DIR'] . "/images/logos/core/login/small_logo_1";
+$smallLogoFinder->files()->in($baseSmallLogoOneDir)->name("logo.*");
+
+if ($smallLogoFinder->hasResults()) {
+    foreach ($smallLogoFinder as $f) {
+        $smallLogoOnePath = $GLOBALS['OE_SITE_WEBROOT'] . "/images/logos/core/login/small_logo_1/" . $f->getFileName();
+    }
+} else {
+    $smallLogoOnePath = $GLOBALS['OE_SITE_WEBROOT'] . "images/logo_1.png";
+}
+
+$smallLogoTwoFinder = new Finder();
+$baseSmallLogoTwoDir = $GLOBALS['OE_SITE_DIR'] . "/images/logos/core/login/small_logo_2";
+$smallLogoTwoFinder->files()->in($baseSmallLogoTwoDir)->name("logo.*");
+
+if ($smallLogoTwoFinder->hasResults()) {
+    foreach ($smallLogoTwoFinder as $f) {
+        $smallLogoTwoPath = $GLOBALS['OE_SITE_WEBROOT'] . "/images/logos/core/login/small_logo_2/" . $f->getFileName();
+    }
+} else {
+    $smallLogoTwoPath = $GLOBALS['OE_SITE_WEBROOT'] . "images/logo_2.png";
+}
```

Comment:
Would be ideal to have a Logo class that encapsulated above, so could then grab the paths for each logo with a one liner that could then be used in other places (for example the patient portal and the oauth2 client page thingy).

Reply:
Done